sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruningSuite.scala

```diff
@@ -85,10 +90,95 @@ class ParquetSchemaPruningSuite
     briefContacts.map { case BriefContact(id, name, address) =>
       BriefContactWithDataPartitionColumn(id, name, address, 2) }
 
+  testSchemaPruning("testing bug") {
+    val data = sql("select * from contacts")
+    import data.sqlContext.implicits._
+
+    val firstAndLastName = udf((first: String, last: String) => first + " " + last)
+
+    data.show(10)
+
+    val query = data.as[Contact]
+      .map(c => c.copy(id = 2))
+      .select("id", "name", "address", "friends")
+      .select(col("id"), firstAndLastName(col("name.first"), col("name.last")))
+
+    query.explain(true)
+    query.show(true)
+
+    val a =10
```

Comment:
Could you remove this line (which is failing the ScalaStyle test)? Then, we can see UT test result.